libraries/script-engine/src/ScriptEngine.cpp

```diff
@@ -1152,6 +1274,54 @@ QObject* ScriptEngine::setupTimerWithInterval(const QScriptValue& function, int
     return newTimer;
 }
 
+void ScriptEngine::processNextImmediate() {
+    if (!_immediateQueue.isEmpty()) {
+        qCDebug(scriptengine) << "processing next immediate (queue length #" << _immediateQueue.size() << ")";
+        ImmediateData *immediateData = _immediateQueue.dequeue();
+        ImmediateData &immediate = *immediateData;
+        if (!immediate.definingEntityIdentifier.isNull() && !isEntityScriptRunning(immediate.definingEntityIdentifier)) {
+            qCDebug(scriptengine) << "ignoring immediate for no longer running entityID" << immediate.definingEntityIdentifier;
+        } else {
+            qCDebug(scriptengine) << "executing immediate (entityID: " << immediate.definingEntityIdentifier << ")";
+            EntityScriptDetails details = _entityScripts[immediate.definingEntityIdentifier];
+            QScriptValue entityScript = details.scriptObject; // previously loaded
+            // call the associated JS function, if it exists
+            if (immediate.function.isValid()) {
+                qCDebug(scriptengine) << "executing immediate.function (js)";
+                callWithEnvironment(immediate.definingEntityIdentifier, immediate.definingSandboxURL, immediate.function, entityScript, QScriptValueList());
+            }
+            std::function<void()> operation = immediate.operation;
+            if (operation) {
+                qCDebug(scriptengine) << "executing immediate.operation (C++)";
+                doWithEnvironment(immediate.definingEntityIdentifier, immediate.definingSandboxURL, operation);
+            }
+        }
+        delete immediateData;
+    }
+}
+
+QObject* ScriptEngine::setImmediate(const QScriptValue& function) {
+    if (DependencyManager::get<ScriptEngines>()->isStopped()) {
+        scriptWarningMessage("Script.setImmediate() while shutting down is ignored... parent script:" + getFilename());
+        return nullptr; // bail early
+    }
+    ImmediateData *immediateData = new ImmediateData(currentEntityIdentifier, currentSandboxURL, function);
+    _immediateQueue.enqueue(immediateData);
+    qDebug() << "setImmediate" << immediateData->definingEntityIdentifier << "(queue #" << _immediateQueue.length() << ")";
+    return immediateData;
+}
+
+bool ScriptEngine::clearImmediate(QObject *data) {
+    ImmediateData* immediateData = reinterpret_cast<ImmediateData*>(data);
+    if (_immediateQueue.contains(immediateData)) {
+        _immediateQueue.removeAll(immediateData);
+        qDebug() << "clearImmediate" << immediateData->definingEntityIdentifier << "(queue #" << _immediateQueue.length() << ")";
+        delete immediateData;
+        return true;
+    }
+    return false;
+}
+
```

Comment:
... like [developer.mozilla.org/Web/API/Window/setImmediate](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate).

Might be overkill to expose to scripting, but `require` already demanded a way to more precisely schedule order-of-operations (for module scoping, caching and exporting).

Also thinking `setImmediate` could offer a good compromise for scripts wanting to be courteous, but with no particular interest in dealing in milliseconds.  Basically way to say "run this callback ASAP and in the queued order" (instead of calculating a future moment and "run this callback exactly at the specified time").